activation/handler_v1_test.go

```diff
@@ -303,14 +303,24 @@ func TestHandlerV1_SyntacticallyValidateAtx(t *testing.T) {
 	})
 	t.Run("can't find VRF nonce", func(t *testing.T) {
 		t.Parallel()
-		atxHdlr, prevATX, postAtx := setup(t)
+		atxHdlr, prevPrevAtx, posAtx := setup(t)
 
-		atx := newChainedActivationTxV1(t, goldenATXID, prevATX, postAtx.ID())
+		// The prevAtx has no VRF nonce in the blob
+		prevAtx := newChainedActivationTxV1(t, goldenATXID, prevPrevAtx, posAtx.ID())
+		prevAtx.Sign(sig)
+		atxHdlr.expectAtxV1(prevAtx, sig.NodeID())
+		atxHdlr.mockFetch.EXPECT().GetAtxs(gomock.Any(), gomock.Any(), gomock.Any())
+		_, err := atxHdlr.processATX(context.Background(), "", prevAtx, codec.MustEncode(prevAtx), time.Now())
+		require.NoError(t, err)
+
+		// ATX increases the number of units - it will need to reverify the existing VRF nonce...
+		atx := newChainedActivationTxV1(t, goldenATXID, prevAtx, prevAtx.ID())
 		atx.NumUnits += 100
 		atx.Sign(sig)
 
+		// .. which is not found in the DB or the previous ATX blob
 		enc := func(stmt *sql.Statement) { stmt.BindBytes(1, atx.SmesherID.Bytes()) }
-		_, err := atxHdlr.cdb.Exec(`UPDATE atxs SET nonce = NULL WHERE pubkey = ?1;`, enc, nil)
+		_, err = atxHdlr.cdb.Exec(`UPDATE atxs SET nonce = NULL WHERE pubkey = ?1;`, enc, nil)
 		require.NoError(t, err)
```

Comment:
This is not a valid test in my opinion - with the change there will never be an ATX in the DB where `nonce` is NULL. Instead update the test where the old VRF nonce fails validation against the new size of the PoST.

Reply:
Yes, `nonce` column should never be `NULL`. Unlike the contents of the blob its value is derived from prev atx in case if it's not specified

Reply:
The test covers error handling in https://github.com/spacemeshos/go-spacemesh/blob/92997020028009f162206a2d4c2ba32b006cb042/activation/handler_v1.go#L254-L257 How do you propose to test it differently?

Reply:
But this is something that should never happen in a real node - we could theoretically enforce it even by adding the `NOT NULL` constraint to the `nonce` column of the table.

Reply:
It doesn't answer the question :)

> we could theoretically enforce it even by adding the NOT NULL constraint to the nonce column of the table.

Yeah, that's what I plan to do (I'm in the process of creating an issue for it atm).

Reply:
> It doesn't answer the question :)

Not test these 3 lines of code specifically for failure - as such a failure can't occur in a real node due to a `sql.ErrNotFound` and a failure because of another error is hard to test for (and we don't do either in other sections of the codebase).

Reply:
The fact that we don't test error handling already caused some troubles, so I don't think it's a good suggestion. Also, the fact that error handling is not tested in other places doesn't mean it shouldn't be tested anywhere.

Reply:
But the test checks something that can never happen?

I think the problem I have with the test and the code it tests is that we are checking a value that cannot be nil if it is?

Reply:
Removed the test :shrug: